Assets/Scripts/UI/PauseMenu/WorldSpaceOverlayUI.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace LekAanDek.UI
+{
+    /// <summary>
+    /// This script is used to render UI infront of everything
+    /// </summary>
+    [ExecuteInEditMode] //Disable if you don't care about previewing outside of play mode
+    public class WorldSpaceOverlayUI : MonoBehaviour
+    {
+        private const string _shaderTestMode = "unity_GUIZTestMode"; //The magic property we need to set
+
+        [SerializeField]
+        public UnityEngine.Rendering.CompareFunction desiredUIComparison = UnityEngine.Rendering.CompareFunction.Always; //If you want to try out other effects
+
+        [Tooltip("Set to blank to automatically populate from the child UI elements")]
+        [SerializeField]
```

Comment:
Why are you serializing public variables? Either make them private, or remove serialization.